spacelift/resource_named_webhook_secret_header.go

```diff
@@ -0,0 +1,152 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func resourceNamedWebhookSecretHeader() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_named_webhook_secret_header` represents secret key value combination used as a custom header" +
+			"when delivering webhook requests. It depends on `spacelift_named_webhook` resource which should exist.",
+
+		CreateContext: resourceNamedWebhookSecretHeaderCreate,
+		ReadContext:   resourceNamedWebhookSecretHeaderRead,
+		DeleteContext: resourceNamedWebhookSecretHeaderDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"webhook_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack on which the environment variable is defined",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"key": {
+				Type:             schema.TypeString,
+				Description:      "key for the header",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+				ForceNew:         true,
+			},
+			"value": {
+				Type:             schema.TypeString,
+				Description:      "value for the header",
+				DiffSuppressFunc: suppressValueChange,
+				Required:         true,
+				Sensitive:        true,
```

Comment:
value can be an empty string?

Reply:
@michieldewilde, an empty string is a valid header value according to RFC 7230

While I would guess that almost no one will ever do that, it's better to allow this, rather than add artificial restrictions.